src/session_impl.cpp

```diff
@@ -6214,9 +6216,14 @@ namespace {
 		m_dht->get_peers(info_hash, std::bind(&on_dht_get_peers, std::ref(m_alerts), info_hash, _1));
 	}
 
-	void session_impl::dht_announce(sha1_hash const& info_hash, int port, dht::announce_flags_t const flags)
+	void session_impl::dht_announce(sha1_hash const& info_hash, int port, dht::announce_flags_t flags)
 	{
 		if (!m_dht) return;
+		if (const auto announce_port = std::uint16_t(m_settings.get_int(settings_pack::announce_port)))
+		{
+			port = announce_port;
+			flags &= ~dht::announce::implied_port;
+		}
```

Comment:
is this change really necessary? you already have logic to not set `implied_port` and to pass in the overridden announce port, in `torrent.cpp` line 2008

Reply:
I believe so? Based on my understanding `dht_tracker::announce` is called directly from both `torrent::dht_announce` as well as `session_impl::dht_announce`. Hence, I added the `announce_port` in both of these places.

I don't believe there is a relationship between `torrent::dht_announce` and `session_impl::dht_announce` where one would affect the other?

Reply:
I see

Reply:
as far as I can tell, this function is only called from `session_handle.cpp`, which means the *user* is calling it. It might be a bit unintuitive to override the port and flag passed in directly from the client.

Reply:
Alright so ideally we would have the `announce_port` setting only affect announcements which are not _client_-issued?

If so, would that mean I roll back this change and only keep `torrent::dht_announce` for DHT and `session_impl::queue_tracker_request` for HTTP/UDP?

In which case it might also be worth documenting the non-application of the `announce_port` on the `session_handle`.
https://github.com/arvidn/libtorrent/blob/7f691246aa58d756fbcdc2f3a03b1623055148ac/include/libtorrent/session_handle.hpp#L502-L517

Thanks for your patience!

Reply:
yeah, I think that sounds good